swarm/storage/types.go

```diff
@@ -105,14 +107,14 @@ func IsZeroKey(key Key) bool {
 
 var ZeroKey = Key(common.Hash{}.Bytes())
 
-func MakeHashFunc(hash string) SwarmHasher {
+func MakeHashFunc(hash string) func() swarmhash.Hash {
 	switch hash {
 	case "SHA256":
-		return func() SwarmHash { return &HashWithLength{crypto.SHA256.New()} }
+		return func() swarmhash.Hash { return &HashWithLength{crypto.SHA256.New()} }
 	case "SHA3":
-		return func() SwarmHash { return &HashWithLength{sha3.NewKeccak256()} }
+		return func() swarmhash.Hash { return &HashWithLength{sha3.NewKeccak256()} }
 	case "BMT":
-		return func() SwarmHash {
+		return func() swarmhash.Hash {
```

Comment:
if anything the BMT should use the SHA3 hashing service. BMT itself not sure if benefits from this worker thing. have you checked somehow?

Reply:
Does the benchmark mentioned in the PR text cover it?

https://github.com/ethersphere/go-ethereum/issues/427